test/unit/finance/no_variable_cost_test.rb

```diff
@@ -1,16 +1,51 @@
 require 'test_helper'
 
 class Finance::NoVariableCostClass < ActiveSupport::TestCase
+  attr_reader :contract, :account, :app_plan
 
-  def test_bill_for_variable
-    contract = FactoryBot.build_stubbed(:contract)
-    account  = FactoryBot.build_stubbed(:simple_account)
-    app_plan = FactoryBot.build_stubbed(:application_plan)
+  setup do
+    @contract = FactoryBot.build_stubbed(:contract)
+    @account  = FactoryBot.build_stubbed(:simple_account)
+    @app_plan = FactoryBot.build_stubbed(:application_plan)
+
+    @org_tz = ENV["TZ"]
+  end
+
+  teardown do
+    ENV["TZ"] = @org_tz
+  end
+
+  test "bill for variable" do
+    Timecop.travel(15.days.ago) if Time.now.mday == 1
 
     contract.stubs(:provider_account).returns(account)
     account.stubs(:provider_can_use?).returns(true)
 
     contract.expects(:save).returns(true)
     contract.notify_observers(:bill_variable_for_plan_changed, app_plan)
   end
+
+  test "no variable billing on 1st day of month local time" do
+    # this changes server and client local time to China Standard Time (+8)
+    ENV["TZ"] = "Asia/Shanghai"
```

Comment:
Tthe test makes sure that billing depends on timezone. If somebody changes `#instant_bill_plan_change` to use `Time.zone.now` instead of the current `Time.now` the test will fail.

Presently no test will fail by such a change.

That is - make sure we retain current behavior until we really want it changed.

I chose this timezone as it is stable 8+ hours AFAIK no DTS but ±1 hour will not affect test outcome. If we rely on local machine timezone, I don't see how to properly test this.

Reply:
OK so you want it to be timezone aware ...
You can use `Time.zone = 'Asia/Shanghai'` see https://api.rubyonrails.org/v5.0/classes/ActiveSupport/TimeZone.html

Reply:
Oh I get it now why you use the environment variable ENV['TZ']
It is because of local time.